dashboard/src/actions/tocActions.js

```diff
@@ -41,6 +45,20 @@ export const fetchTOC =
     dispatch({ type: TYPES.COMPLETED });
   };
 
+const setOptions = (data, isParent, keyPath, isDirectory) => {
+  const options = data.map((item) => ({
+    name: item.name,
+    id: isParent ? `${keyPath}*${item.name}` : item.name,
+    isDirectory,
+    uri: item.uri,
+  }));
+  if (isDirectory) {
+    options.forEach((opt) => {
+      opt["children"] = [];
+    });
+  }
```

Comment:
Interesting. The right pane tries to show size, referencing `activeFile.size`, which presumably succeeds, but seems to produce no value:

![image](https://github.com/distributed-system-analysis/pbench/assets/61761529/536eabff-0854-4aa9-8de6-3f459b96210c)

So how is the `size` field even defined?

We need to copy it from `data` inside the map lambda: it's one thing to arbitrarily create empty `children` nodes on the entire list of `options` directories, but that won't work for size. (And we might as well just create `children` inside the lambda, too, instead of traversing `options` again to do it after the fact.)

Reply:
Before this most recent update, the code in this PR was fetching the size from each entry in `contentData.files`.  In the refactoring for this latest update, the commonalty between the directory handling and the file handling was moved to this `setOptions()` function; the special handling for directories got moved too, but the file-specific stuff got left behind.

I'm not sure that there is much point in putting the `children` creation inside the same traversal as the `size` addition, since, for a given call, there will only be directories or files but never both -- i.e., `isDirectory` is either `true` or `false` -- but, if it's `false`, we can do a separate traversal for the files-handling.